src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -781,6 +785,57 @@ public boolean isAvailable()
                 getJDA().getEntityBuilder().createMember(this, resp.getObject()));
     }
 
+    @Nonnull
+    @Override
+    public CompletableFuture<MemberPresence> retrieveMemberPresence(long memberId)
+    {
+        Member member = getMemberById(memberId);
+        if (member != null
+            && api.isCacheFlagSet(CacheFlag.ACTIVITY)
+            && api.isCacheFlagSet(CacheFlag.CLIENT_STATUS))
+            return CompletableFuture.completedFuture(new MemberPresenceImpl(member));
+
+        CompletableFuture<MemberPresence> future = new CompletableFuture<>();
+        if (memberId == 0)
+        {
+            future.completeExceptionally(new IllegalArgumentException("Member with the specified userId does not exist"));
+            return future;
+        }
+
+        api.getClient().requestMembers((response) -> {
```

Comment:
Let's make the callback the last argument to make this more readable

Reply:
I had it this way because the `userId` parameter was meant as a varargs to allow requesting multiple users. I'd rather keep it like this for future proofing.

Reply:
This is an internal method, so having it be a single parameter and later changing that to am array should be fine. We needn't build in future proofing for that